ClientSupport/src/test/java/io/deephaven/clientsupport/gotorow/SeekRowTest.java

```diff
@@ -0,0 +1,307 @@
+        assertNaturallySorted(
+                new int[] {1, 1, 1, 2}, 1,
+                new int[] {1, 2, 0, 0},
+                new int[] {2, 0, 1, 2},
+                new int[] {1, 2, 3, 1},
+                new int[] {3, 3, 1, 2});
+    }
+
+    @Test
+    public void middle1() {
+        assertNaturallySorted(
+                new int[] {1, 2, 3}, 2,
+                new int[] {1, 1, 1},
+                new int[] {1, 1, 1},
+                new int[] {1, 1, 1},
+                new int[] {1, 1, 1});
+    }
+
+    @Test
+    public void middle2() {
+        assertNaturallySorted(
+                new int[] {1, 2, 2, 3}, 2,
+                new int[] {1, 2, 1, 1},
+                new int[] {2, 2, 1, 2},
+                new int[] {1, 2, 1, 1},
+                new int[] {2, 2, 1, 2});
+    }
+
+    @Test
+    public void middle3() {
+        assertNaturallySorted(
+                new int[] {1, 2, 2, 2, 3}, 2,
+                new int[] {1, 2, 3, 1, 1},
+                new int[] {3, 3, 1, 2, 3},
+                new int[] {1, 2, 3, 1, 1},
+                new int[] {3, 3, 1, 2, 3});
+    }
+
+    @Test
+    public void end1() {
+        assertNaturallySorted(
+                new int[] {1, 2}, 2,
+                new int[] {1, 1},
+                new int[] {1, 1},
+                new int[] {0, 0},
+                new int[] {0, 0});
+    }
+
+    @Test
+    public void end2() {
+        assertNaturallySorted(
+                new int[] {1, 2, 2}, 2,
+                new int[] {1, 2, 1},
+                new int[] {2, 2, 1},
+                new int[] {1, 0, 0},
+                new int[] {1, 0, 1});
+    }
+
+    @Test
+    public void end3() {
+        assertNaturallySorted(
+                new int[] {1, 2, 2, 2}, 2,
+                new int[] {1, 2, 3, 1},
+                new int[] {3, 3, 1, 2},
+                new int[] {1, 2, 0, 0},
+                new int[] {2, 0, 1, 2});
+    }
+
+    @Test
+    public void notFound() {
+        assertNaturallySorted(
+                new int[] {2, 4, 6}, 1,
+                new int[] {-1, -1, -1},
+                new int[] {-1, -1, -1},
+                new int[] {-1, -1, -1},
+                new int[] {-1, -1, -1});
+        assertNaturallySorted(
+                new int[] {2, 4, 6}, 3,
+                new int[] {-1, -1, -1},
+                new int[] {-1, -1, -1},
+                new int[] {-1, -1, -1},
+                new int[] {-1, -1, -1});
+        assertNaturallySorted(
+                new int[] {2, 4, 6}, 5,
+                new int[] {-1, -1, -1},
+                new int[] {-1, -1, -1},
+                new int[] {-1, -1, -1},
+                new int[] {-1, -1, -1});
+        assertNaturallySorted(
+                new int[] {2, 4, 6}, 7,
+                new int[] {-1, -1, -1},
+                new int[] {-1, -1, -1},
+                new int[] {-1, -1, -1},
+                new int[] {-1, -1, -1});
+    }
+
+    @Test
+    public void unsorted() {
```

Comment:
two more things i'd like to see in this test: 
 * find first (and "last when seeking backwards")
 * find in a contiguous region of the same value (so double up one of the `1`s for example, and make sure the "find the next" behaves)